llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -6463,7 +6463,7 @@ getRangeForUnknownRecurrence(const SCEVUnknown *U) {
     // TODO: Handle the power function forms some day.
     return FullSet;
 
-  unsigned TC = getSmallConstantMaxTripCount(L);
+  std::optional<unsigned> TC = getSmallConstantMaxTripCount(L);
   if (!TC || TC >= BitWidth)
```

Comment:
```
TC >= BitWidth
```
It compiles? We compare an optional to a BitWidth?

Reply:
See std::optional<T>::operator>=.